### PR TITLE
DEV-4451: Show canceled_at date in new portal

### PR DESCRIPTION
**Pull request by GuiMend**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
n/a
#### What's this PR do?
- Adds "canceled_at" date to canceled banner in Portal > Contribution Details
#### Why are we doing this? How does it help us?
- Show more useful information to the user regarding cancelation
#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?
yes
#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
yes
#### Have automated unit tests been added? If not, why?
yes
#### How should this change be communicated to end users?
Canceled at date now shown
#### Are there any smells or added technical debt to note?
no
#### Has this been documented? If so, where?
no
#### What are the relevant tickets? Add a link to any relevant ones.
https://news-revenue-hub.atlassian.net/browse/DEV-4451
#### Do any changes need to be made before deployment to production (adding environment variables, for example)? If so, open a ticket and link it to the ticket for this PR and list it here:
no
#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:
no

### CODE REVIEW

**Review by nrh-cklimas — CHANGES_REQUESTED** [review]

File: spa/src/components/portal/ContributionsList/ContributionDetail/Banner/Banner.test.tsx [hunk 1]
```diff
@@ -43,6 +43,34 @@ describe('Banner', () => {
...
+        contribution: { ...mockContribution, status: 'canceled', canceled_at: new Date('1/2/2024').toISOString() }
+      });
+      expect(
+        screen.getByText(
+          'This contribution was canceled at 1/2/2024. Help our community and continue your support of our mission by creating a new contribution.'
+        )
+      ).toBeVisible();
+    });
+
+    it('should render link to donation page', () => {
```
[the remark this answers is not in the document]
Comment (GuiMend) [reply spa/src/components/portal/ContributionsList/ContributionDetail/Banner/Banner.test.tsx:57]: I already have a test `should render description without canceled_at date`, doesn't it cover this scenario?

[the remark this answers is not in the document]
Comment (nrh-cklimas) [reply spa/src/components/portal/ContributionsList/ContributionDetail/Banner/Banner.test.tsx:69]: Ah, I see. Missed it. I think we still need a test around a RP without a default page, though?

File: spa/src/components/portal/ContributionsList/ContributionDetail/Banner/Banner.test.tsx [hunk 2]
```diff
@@ -2,6 +2,12 @@ import { PortalContributionDetail } from 'hooks/usePortalContribution';
 import { axe } from 'jest-axe';
 import { render, screen } from 'test-utils';
 import Banner, { BannerProps } from './Banner';
+import usePortal from 'hooks/usePortal';
+
+jest.mock('hooks/usePortal', () => ({
+  __esModule: true,
+  default: jest.fn()
+}));
```
Comment (nrh-cklimas) [inline spa/src/components/portal/ContributionsList/ContributionDetail/Banner/Banner.test.tsx:10]:
```suggestion
jest.mock('hooks/usePortal');
```

File: spa/src/components/portal/ContributionsList/ContributionDetail/Banner/Banner.test.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (GuiMend) [reply spa/src/components/portal/ContributionsList/ContributionDetail/Banner/Banner.test.tsx:69]: Right. Added one

**Review by nrh-cklimas — APPROVED** [review]

**Pull request merged**